### PR TITLE
Block Processing Sanity Spec Tests

### PR DESCRIPTION
**Pull request by nisdas**
Part of #2812

### CODE REVIEW

File: beacon-chain/core/blocks/spectest/block_processing_test.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	if err != nil {
+		t.Fatalf("Could not load file %v", err)
+	}
+
+	s := &BlocksMinimal{}
+	if err := yaml.Unmarshal(file, s); err != nil {
+		t.Fatalf("Failed to Unmarshal: %v", err)
+	}
+
+	log.Infof("Title: %v", s.Title)
```
Comment (prestonvanloon) [inline beacon-chain/core/blocks/spectest/block_processing_test.go]:
```suggestion
	t.Logf("Title: %v", s.Title)
```

File: beacon-chain/core/blocks/spectest/block_processing_test.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	log.Infof("Summary: %v", s.Summary)
```
Comment (prestonvanloon) [inline beacon-chain/core/blocks/spectest/block_processing_test.go]:
```suggestion
	t.Logf("Summary: %v", s.Summary)
```

File: beacon-chain/core/blocks/spectest/block_processing_test.go [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+	log.Infof("Fork: %v", s.Forks)
```
Comment (prestonvanloon) [inline beacon-chain/core/blocks/spectest/block_processing_test.go]:
```suggestion
	t.Logf("Fork: %v", s.Forks)
```

File: beacon-chain/core/blocks/spectest/block_processing_test.go [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+	log.Infof("Config: %v", s.Config)
```
Comment (prestonvanloon) [inline beacon-chain/core/blocks/spectest/block_processing_test.go]:
```suggestion
	t.Logf("Config: %v", s.Config)
```

File: beacon-chain/core/blocks/spectest/block_processing_test.go [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+	if err != nil {
+		t.Fatalf("Could not load file %v", err)
+	}
+
+	s := &BlocksMainnet{}
+	if err := yaml.Unmarshal(file, s); err != nil {
+		t.Fatalf("Failed to Unmarshal: %v", err)
+	}
+
+	log.Infof("Title: %v", s.Title)
```
Comment (prestonvanloon) [inline beacon-chain/core/blocks/spectest/block_processing_test.go]: Here too, use `t.Logf`

File: shared/params/config.go [hunk 6]
```diff
@@ -273,7 +273,7 @@ func MinimalSpecConfig() *BeaconChainConfig {
 	minimalConfig.SlotsPerHistoricalRoot = 64
 	minimalConfig.MinValidatorWithdrawalDelay = 256
 	minimalConfig.PersistentCommitteePeriod = 2048
-	minimalConfig.MaxEpochsPerCrosslink = 64
+	minimalConfig.MaxEpochsPerCrosslink = 4
```
Comment (prestonvanloon) [inline shared/params/config.go]: Should this be changed?

Comment (prestonvanloon) [reply shared/params/config.go]: I see it here. This looks good.

https://github.com/ethereum/eth2.0-specs/blob/b21c9cc71b4dc33426ed73028731e6a259f43310/configs/constant_presets/minimal.yaml#L68

File: beacon-chain/core/helpers/randao.go [hunk 7]
```diff
@@ -70,6 +70,9 @@ func GenerateSeed(state *pb.BeaconState, epoch uint64) ([32]byte, error) {
 //    """
 //    return state.latest_active_index_roots[epoch % LATEST_ACTIVE_INDEX_ROOTS_LENGTH]
 func ActiveIndexRoot(state *pb.BeaconState, epoch uint64) []byte {
+	newRootLength := len(state.LatestActiveIndexRoots[epoch%params.BeaconConfig().LatestActiveIndexRootsLength])
+	newRoot := make([]byte, newRootLength)
+	copy(newRoot, state.LatestActiveIndexRoots[epoch%params.BeaconConfig().LatestActiveIndexRootsLength])
 	return state.LatestActiveIndexRoots[epoch%params.BeaconConfig().LatestActiveIndexRootsLength]
```
Comment (terencechain) [inline beacon-chain/core/helpers/randao.go]: this still returns a reference, other than that it looks good

**Pull request merged**